engine/renderer/private/resource_management/mesh_resource_manager.cpp

```diff
@@ -0,0 +1,8 @@
+#include "resource_management/mesh_resource_manager.hpp"
+
+#include "gpu_resources.hpp"
+
+template <>
+std::weak_ptr<ResourceManager<Mesh>> ResourceHandle<Mesh>::manager = {};
+
+MeshResourceManager::MeshResourceManager() = default;
```

Comment:
The constructors to each resource manager should not be public. Instead, there should me a Make() static function that returns and assigns a new shared pointer instance as the global manager. This avoid the need to inherit from shared from this and forgetting to assign the global weak pointer to the manager. Personally, I am not fond of this method, but I realize that it is a good way to keep resource handle to 4 bytes